// Validating benchmarks:
// ***** BenchmarkRunner: Start   *****
// ***** Building 1 exe(s) in Parallel: Start   *****
BuildScript: C:\Users\Bastos\Desktop\Exercises Curse C# paralellism\ParallelCsharp\LearnParallel\PartitioningSample\bin\Release\f3116147-fa12-417d-863d-955694cefaad.bat
// ***** Done, took 00:00:02 (2.93 sec)   *****
// Found benchmarks:
//   Program.SquareEachValue: DefaultJob
//   Program.SquareEachValueChuncked: DefaultJob

// **************************
// Benchmark: Program.SquareEachValue: DefaultJob
// *** Execute ***
// Launch: 1 / 1
// Execute: C:\Users\Bastos\Desktop\Exercises Curse C# paralellism\ParallelCsharp\LearnParallel\PartitioningSample\bin\Release\f3116147-fa12-417d-863d-955694cefaad.exe -benchmarkId 0
// BeforeAnythingElse
// Benchmark Process Environment Information:
// Runtime=.NET Framework 4.7.1 (CLR 4.0.30319.42000), 32bit LegacyJIT-v4.7.3130.0
// GC=Concurrent Workstation
// Job: DefaultJob
OverheadJitting  1: 1 op, 294702.39 ns, 294.7024 us/op
System.Reflection.TargetInvocationException: Uma exceção foi acionada pelo destino de uma chamada. ---> System.AggregateException: Um ou mais erros. ---> System.IndexOutOfRangeException: O índice estava fora dos limites da matriz.
   em PartitioningSample.Program.<>c__DisplayClass0_0.<SquareEachValue>b__0(Int32 x) na C:\Users\Bastos\Desktop\Exercises Curse C# paralellism\ParallelCsharp\LearnParallel\PartitioningSample\Program.cs:linha 20
   em System.Threading.Tasks.Parallel.<>c__DisplayClass42_0`2.<PartitionerForEachWorker>b__1()
   em System.Threading.Tasks.Task.InnerInvoke()
   em System.Threading.Tasks.Task.InnerInvokeWithArg(Task childTask)
   em System.Threading.Tasks.Task.<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0(Object )
   --- Fim do rastreamento de pilha de exceções internas ---
   em System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   em System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   em System.Threading.Tasks.Task.Wait()
   em System.Threading.Tasks.Parallel.PartitionerForEachWorker[TSource,TLocal](Partitioner`1 source, ParallelOptions parallelOptions, Action`1 simpleBody, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   em System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   em System.Threading.Tasks.Parallel.ForEach[TSource](IEnumerable`1 source, Action`1 body)
   em PartitioningSample.Program.SquareEachValue() na C:\Users\Bastos\Desktop\Exercises Curse C# paralellism\ParallelCsharp\LearnParallel\PartitioningSample\Program.cs:linha 18
   em BenchmarkDotNet.Autogenerated.Runnable_0.WorkloadActionNoUnroll(Int64 invokeCount)
   em BenchmarkDotNet.Engines.Engine.RunIteration(IterationData data)
   em BenchmarkDotNet.Engines.EngineFactory.Jit(Engine engine, Int32 jitIndex, Int32 invokeCount, Int32 unrollFactor)
   em BenchmarkDotNet.Engines.EngineFactory.CreateReadyToRun(EngineParameters engineParameters)
   em BenchmarkDotNet.Autogenerated.Runnable_0.Run(IHost host)
   --- Fim do rastreamento de pilha de exceções internas ---
   em System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   em System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   em System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   em BenchmarkDotNet.Autogenerated.UniqueProgramName.AfterAssemblyLoadingAttached(String[] args)
// AfterAll
ExitCode != 0
No more Benchmark runs will be launched as NO measurements were obtained from the previous run!


// **************************
// Benchmark: Program.SquareEachValueChuncked: DefaultJob
// *** Execute ***
// Launch: 1 / 1
// Execute: C:\Users\Bastos\Desktop\Exercises Curse C# paralellism\ParallelCsharp\LearnParallel\PartitioningSample\bin\Release\f3116147-fa12-417d-863d-955694cefaad.exe -benchmarkId 1
// BeforeAnythingElse
// Benchmark Process Environment Information:
// Runtime=.NET Framework 4.7.1 (CLR 4.0.30319.42000), 32bit LegacyJIT-v4.7.3130.0
// GC=Concurrent Workstation
// Job: DefaultJob
OverheadJitting  1: 1 op, 312296.57 ns, 312.2966 us/op
System.Reflection.TargetInvocationException: Uma exceção foi acionada pelo destino de uma chamada. ---> System.AggregateException: Um ou mais erros. ---> System.IndexOutOfRangeException: O índice estava fora dos limites da matriz.
   em PartitioningSample.Program.<>c__DisplayClass1_0.<SquareEachValueChuncked>b__0(Tuple`2 range) na C:\Users\Bastos\Desktop\Exercises Curse C# paralellism\ParallelCsharp\LearnParallel\PartitioningSample\Program.cs:linha 33
   em System.Threading.Tasks.Parallel.<>c__DisplayClass42_0`2.<PartitionerForEachWorker>b__1()
   em System.Threading.Tasks.Task.InnerInvoke()
   em System.Threading.Tasks.Task.InnerInvokeWithArg(Task childTask)
   em System.Threading.Tasks.Task.<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0(Object )
   --- Fim do rastreamento de pilha de exceções internas ---
   em System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   em System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   em System.Threading.Tasks.Task.Wait()
   em System.Threading.Tasks.Parallel.PartitionerForEachWorker[TSource,TLocal](Partitioner`1 source, ParallelOptions parallelOptions, Action`1 simpleBody, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   em System.Threading.Tasks.Parallel.ForEach[TSource](Partitioner`1 source, Action`1 body)
   em PartitioningSample.Program.SquareEachValueChuncked() na C:\Users\Bastos\Desktop\Exercises Curse C# paralellism\ParallelCsharp\LearnParallel\PartitioningSample\Program.cs:linha 31
   em BenchmarkDotNet.Autogenerated.Runnable_1.WorkloadActionNoUnroll(Int64 invokeCount)
   em BenchmarkDotNet.Engines.Engine.RunIteration(IterationData data)
   em BenchmarkDotNet.Engines.EngineFactory.Jit(Engine engine, Int32 jitIndex, Int32 invokeCount, Int32 unrollFactor)
   em BenchmarkDotNet.Engines.EngineFactory.CreateReadyToRun(EngineParameters engineParameters)
   em BenchmarkDotNet.Autogenerated.Runnable_1.Run(IHost host)
   --- Fim do rastreamento de pilha de exceções internas ---
   em System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   em System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   em System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   em BenchmarkDotNet.Autogenerated.UniqueProgramName.AfterAssemblyLoadingAttached(String[] args)
// AfterAll
ExitCode != 0
No more Benchmark runs will be launched as NO measurements were obtained from the previous run!


// ***** BenchmarkRunner: Finish  *****

// * Export *
  BenchmarkDotNet.Artifacts\results\PartitioningSample.Program-report.csv
  BenchmarkDotNet.Artifacts\results\PartitioningSample.Program-report-github.md
  BenchmarkDotNet.Artifacts\results\PartitioningSample.Program-report.html

// * Detailed results *
Program.SquareEachValue: DefaultJob
Runtime = ; GC = 
There are not any results runs

Program.SquareEachValueChuncked: DefaultJob
Runtime = ; GC = 
There are not any results runs

// * Summary *

BenchmarkDotNet=v0.11.0, OS=Windows 8.1 (6.3.9600.0)
Intel Core i7-3612QM CPU 2.10GHz (Ivy Bridge), 1 CPU, 8 logical and 4 physical cores
Frequency=2046132 Hz, Resolution=488.7270 ns, Timer=TSC
  [Host] : .NET Framework 4.7.1 (CLR 4.0.30319.42000), 32bit LegacyJIT-v4.7.3130.0


                  Method | Mean | Error |
------------------------ |-----:|------:|
         SquareEachValue |   NA |    NA |
 SquareEachValueChuncked |   NA |    NA |

Benchmarks with issues:
  Program.SquareEachValue: DefaultJob
  Program.SquareEachValueChuncked: DefaultJob

// * Warnings *
Environment
  Summary -> Detected error exit code from one of the benchmarks. It might be caused by following antivirus software:
        - Windows Defender (%ProgramFiles%\Windows Defender\MSASCui.exe)
Use InProcessToolchain to avoid new process creation.


// * Legends *
  Mean  : Arithmetic mean of all measurements
  Error : Half of 99.9% confidence interval
  1 ns  : 1 Nanosecond (0.000000001 sec)

// ***** BenchmarkRunner: End *****
Run time: 00:00:00 (0.73 sec), executed benchmarks: 2

// * Artifacts cleanup *
